src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/Tabs.js

```diff
@@ -1,56 +1,308 @@
+        const {small} = this.props;
+
+        return React.Children.map(tabs, (tab, localIndex) => {
+            const index = indices[localIndex];
+
             if (!tab) {
                 return null;
             }
 
+            const selected = this.isSelected(index);
+
             return React.cloneElement(
                 tab,
                 {
                     ...tab.props,
                     index,
-                    selected: this.isSelected(index),
-                    onClick: this.handleSelectionChange,
+                    selected,
+                    small,
+                    onClick: this.handleTabClick,
+                    setWidth: this.setTabWidth,
                 }
             );
         });
     }
 
-    handleSelectionChange = (selectedTabIndex: ?number) => {
-        if (typeof(selectedTabIndex) !== 'undefined' && selectedTabIndex !== null) {
-            this.props.onSelect(selectedTabIndex);
-        }
-    };
+    createCollapsedTabItems(tabs: Array<Element<typeof Tab> | false>, indices: number[]) {
+        const {skin} = this.props;
+
+        return React.Children.map(tabs, (tab, localIndex) => {
+            const index = indices[localIndex];
+
+            if (!tab) {
+                return null;
+            }
+
+            return (
+                <CollapsedTab
+                    index={index}
+                    key={index}
+                    onClick={this.handleCollapsedTabClick}
+                    skin={tab.props.skin || skin}
+                >
+                    {tab.props.children}
+                </CollapsedTab>
+            );
+        });
+    }
 
     render() {
         const {
-            children,
+            skin,
+            small,
         } = this.props;
-        const tabsItems = this.createTabItems(children);
+
+        const containerClass = classNames(
+            tabsStyles.container,
+            tabsStyles[skin],
+            {
+                [tabsStyles.small]: small,
+            }
+        );
 
         return (
-            <ul className={tabsStyles.tabs}>
-                {tabsItems}
-            </ul>
+            <div className={containerClass} ref={this.setContainerRef}>
+                <div className={tabsStyles.wrapper} ref={this.setWrapperRef}>
+                    <ul className={tabsStyles.tabs}>
+                        {this.visibleTabs}
+                    </ul>
+                </div>
+
+                {this.hasHiddenTabs &&
+                    <React.Fragment>
+                        <button
+                            className={tabsStyles.button}
+                            onClick={this.handleDropdownToggle}
+                        >
+                            <Icon name="su-more-horizontal" />
+                        </button>
+
+                        <Popover
+                            anchorElement={this.containerRef || undefined}
+                            horizontalOffset={10000}
```

Comment:
I don't really understand your suggestion. AFAIK the popover will always render left-aligned as long as there is enough space on the right side of the anchor. So I probably can't really do something else.

Reply:
What happens if you make the `button` the `anchorElement`? Since the content of the `Popover` is probably always wider than the `button` it would still be right aligned. Your solution breaks as soon as we have screens with such a high resolution.

What I am suggesting would look something like this (not working, just sketching an idea):

```javascript
                    <button
                        className={buttonClass}
                        onClick={this.handleDropdownToggle}
                        ref={this.setButtonRef}
                    >
                        <Icon name="su-more-horizontal" />
                    </button>

                    {this.hasCollapsedTabs &&
                        <Popover
                            anchorElement={this.buttonRef || undefined}
                            onClose={this.handleDropdownClose}
                            open={this.dropdownOpen}
                        >
                        </Popover>
```